src/text/parsers/integer.rs

```diff
@@ -88,28 +92,63 @@ fn base_2_integer_digits(input: &str) -> IResult<&str, &str> {
 fn base_10_integer(input: &str) -> IResult<&str, TextValue> {
     map_res(
         recognize(preceded(opt(char('-')), base_10_integer_digits)),
-        |text| parse_i64_with_radix(text, 10).map(TextValue::Integer),
+        |text| parse_integer_with_radix(text, 10).map(TextValue::Integer),
     )(input)
 }
 
 /// Strips any underscores out of the provided text and then parses it according to the specified
 /// radix.
-fn parse_i64_with_radix(text: &str, radix: u32) -> Result<i64, ParseIntError> {
+fn parse_integer_with_radix(text: &str, radix: u32) -> IonResult<Integer> {
```

Comment:
PR tour: the `RawTextReader` had to be modified to return a `BigInt` if the int being processed was too large to fit in an `i64`.